pageserver/src/page_service.rs

```diff
@@ -273,10 +273,28 @@ async fn page_service_conn_main(
                 info!("Postgres client disconnected ({io_error})");
                 Ok(())
             } else {
-                Err(io_error).context("Postgres connection error")
+                let tenant_id = conn_handler
+                    .timeline_handles
+                    .tenant_id()
+                    .map(|tid| format!("{tid}"))
+                    .unwrap_or("???".to_string());
+                Err(io_error).context(format!(
+                    "Postgres connection error for tenant {} client at {}",
+                    tenant_id, peer_addr
+                ))
             }
         }
-        other => other.context("Postgres query error"),
+        other => {
+            let tenant_id = conn_handler
+                .timeline_handles
+                .tenant_id()
+                .map(|tid| format!("{tid}"))
+                .unwrap_or("???".to_string());
```

Comment:
I wonder if it would make sense to just fix TenantId and TimelineId to have the same display and debug forms (debug calls display). Then we could just use `{:?}` with the `conn_handler.tenant_id()` and it would receive automatic `Some(<tenant_id>)` or `None` formatting.

Reply:
I think it's already the case: https://github.com/neondatabase/neon/blob/main/libs/utils/src/id.rs#L228-L238.
Good suggestion. Done in https://github.com/neondatabase/neon/pull/9130/commits/fd81a6105972bf58226a63e3681b41d28d072afa

Reply:
Oki, I did check it before writing the comment and must've just misread there being a derive. Excellent.